### ISSUE TITLE
`consistent-function-scoping` false positive with an arrow function using `this`

### ISSUE THREAD

**Issue by yvele** [sindresorhus/eslint-plugin-unicorn#2088]
With the following rule:

```js
"unicorn/consistent-function-scoping" : ["error", {
  checkArrowFunctions : true
}]
```

(I do really want arrow function to move upper scope when they do not use any variable or upper `this`).

This is wrongly reported as a problem:

```ts
class Foo {
  public bar = new FinalizationRegistry(() => {
    console.log(this); // <- `this` is used in this arrow function
  })
}
```

> error  Move arrow function to the outer scope  unicorn/consistent-function-scoping

Same for:

```js
function foo(this) {
  const bar = () => console.log(this); // <- `this` is used in this arrow function
  return [].map(() => bar);
}
```

> error  Move arrow function 'bar' to the outer scope

The cases below should be OK when an arrow function only consuming `this` from outer scope.

**Comment by silverwind** [comment]
Just had this false-positive after switch eslint's parser to `@typescript-eslint/parser`. The bug does not happen with the default `espree` parser.

**Comment by Samuel-Therrien-Beslogic** [comment]
I think this (false-positive?) in an Angular context is the same symptom:
```ts
@Directive({
  selector: '[appTooltip]',
  standalone: true,
})
export class TooltipDirective<THtmlElement extends HTMLElement>
{
  // eslint-disable-next-line unicorn/consistent-function-scoping -- Uses this
  private readonly _describeBy = computed(() => {
    const tooltipDescription = this.tooltipDescription()
    if (tooltipDescription) return tooltipDescription
    const appTooltip = this.appTooltip()

    return (typeof appTooltip === 'string')
      ? beslogicTooltip
      : ''
  })

  // ...
```

Where it can be "fixed" this way:
```ts
@Directive({
  selector: '[appTooltip]',
  standalone: true,
})
export class TooltipDirective<THtmlElement extends HTMLElement>
{
  private readonly _compute = () => {
    const tooltipDescription = this.tooltipDescription()
    if (tooltipDescription) return tooltipDescription
    const appTooltip = this.appTooltip()

    return (typeof appTooltip === 'string')
      ? beslogicTooltip
      : ''
  }

  private readonly _describeBy = computed(_compute)

  // ...
```

But this doesn't seem useful as you end up constructing the function on every instantiation either way.
